src/measure_theory/interval_integral.lean

```diff
@@ -1381,6 +1381,12 @@ theorem integral_deriv_eq_sub (hderiv : ∀ x ∈ interval a b, differentiable_a
   ∫ y in a..b, deriv f y = f b - f a :=
 integral_eq_sub_of_has_deriv_at (λ x hx, (hderiv x hx).has_deriv_at) hcont'
 
+theorem integral_deriv_eq_sub' (f) (hderiv : deriv f = f')
```

Comment:
```suggestion
theorem integral_deriv_eq_sub' {f} (hderiv : deriv f = f')
```

Reply:
That's what I had done originally, but it turns out that there are some instances (`integral_sin`, for example) where it is much easier to pass the function as an argument and then let `simp` handle `deriv f = f'` rather than to prove `deriv f = f'` outright. Making `f` explicit here is an easy fix.

Reply:
(It needs to be fed the function in those instances)

Reply:
I really dislike this `{x} (h : p x)` pattern, where the term is implicitly derived from the proof.  One reason, like Benjamin noted, is that it is awkward to use with simp, linarith, or really any automation.  Another reason is that it often makes it harder to write top-down proofs: if you write `foo _` and Lean can't unify the conclusion of the lemma with the goal, you're lost.